### PR TITLE
Upgrade Protobuf to 3.13.0

### LINKED ISSUE #67: Compilation failure with grpc-java
I tried upgrading grpc-java in my project from 1.29.0 to 1.31.0, but received this compilation error:

```
ERROR: /home/admin/.cache/bazel/_bazel_admin/b812dc62599757b102340939b9116fb8/external/io_grpc_grpc_java/compiler/BUILD.bazel:5:1: C++ compilation of rule '@io_grpc_grpc_java//compiler:grpc_java_plugin' failed (Exit 1) gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining 41 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
external/io_grpc_grpc_java/compiler/src/java_plugin/cpp/java_plugin.cpp:46:12: error: 'uint64_t JavaGrpcGenerator::GetSupportedFeatures() const' marked 'override', but does not override
   uint64_t GetSupportedFeatures() const override {
            ^~~~~~~~~~~~~~~~~~~~
external/io_grpc_grpc_java/compiler/src/java_plugin/cpp/java_plugin.cpp: In member function 'uint64_t JavaGrpcGenerator::GetSupportedFeatures() const':
external/io_grpc_grpc_java/compiler/src/java_plugin/cpp/java_plugin.cpp:47:12: error: 'FEATURE_PROTO3_OPTIONAL' was not declared in this scope
     return FEATURE_PROTO3_OPTIONAL;
            ^~~~~~~~~~~~~~~~~~~~~~~
```

I _think_ this is because we need at least protobuf 3.12.0 now, but the rules seem to hardcode 3.11.3:

https://github.com/bazelbuild/rules_proto/blob/40298556293ae502c66579620a7ce867d5f57311/proto/private/dependencies.bzl#L29

https://github.com/grpc/grpc-java/releases/tag/v1.30.0
> Bumped protobuf to 3.12.0

Is it possible to upgrade the protobuf version, or allow it to be passed in?

### PR DESCRIPTION
**Pull request by Yannic**
Fixes #67

### CODE REVIEW

**Comment by Yannic** [comment]
@meteorcloudy Can you please upload these files to bazel-mirror? Thank you!

```
  https://github.com/protocolbuffers/protobuf/archive/v3.13.0.tar.gz
  https://github.com/protocolbuffers/protobuf/releases/download/v3.13.0/protoc-3.13.0-linux-aarch_64.zip
  https://github.com/protocolbuffers/protobuf/releases/download/v3.13.0/protoc-3.13.0-linux-ppcle_64.zip
  https://github.com/protocolbuffers/protobuf/releases/download/v3.13.0/protoc-3.13.0-linux-s390x_64.zip
  https://github.com/protocolbuffers/protobuf/releases/download/v3.13.0/protoc-3.13.0-linux-x86_32.zip
  https://github.com/protocolbuffers/protobuf/releases/download/v3.13.0/protoc-3.13.0-linux-x86_64.zip
  https://github.com/protocolbuffers/protobuf/releases/download/v3.13.0/protoc-3.13.0-osx-x86_32.zip
  https://github.com/protocolbuffers/protobuf/releases/download/v3.13.0/protoc-3.13.0-osx-x86_64.zip
  https://github.com/protocolbuffers/protobuf/releases/download/v3.13.0/protoc-3.13.0-win32.zip
  https://github.com/protocolbuffers/protobuf/releases/download/v3.13.0/protoc-3.13.0-win64.zip
  https://repo1.maven.org/maven2/com/google/protobuf/protobuf-java/3.13.0/protobuf-java-3.13.0.jar
  https://repo1.maven.org/maven2/com/google/protobuf/protobuf-java/3.13.0/protobuf-java-3.13.0-sources.jar
  https://repo1.maven.org/maven2/com/google/protobuf/protobuf-java-util/3.13.0/protobuf-java-util-3.13.0.jar
  https://repo1.maven.org/maven2/com/google/protobuf/protobuf-java-util/3.13.0/protobuf-java-util-3.13.0-sources.jar
  https://repo1.maven.org/maven2/com/google/protobuf/protobuf-javalite/3.13.0/protobuf-javalite-3.13.0.jar
  https://repo1.maven.org/maven2/com/google/protobuf/protobuf-javalite/3.13.0/protobuf-javalite-3.13.0-sources.jar
```

**Comment by meteorcloudy** [comment]
@Yannic It's done!

**Comment by Yannic** [comment]
TODO(yannic): Upgrade version shipped with Bazel after this has landed.

**Comment by Yannic** [comment]
@comius @lberki Friendly ping?

File: proto/private/generate_sums.sh
Comment (eklitzke) [inline proto/private/generate_sums.sh:2]: nit: I see why you might want to remove the shebang line here as this hardcodes the path to bash to the location on macos, but I don't think this is right. For one thing you've left the file as executable, so if the sheband is removed the executable bit should be removed as well. I don't have a mac but afaik there's a symlink from /bin/bash on macos, or worst case you could do something like `#!/usr/bin/env bash`

**Review by eklitzke — DISMISSED** [review]
I tested this branch locally in a project of mine that uses rules_proto and it worked for me.

File: proto/private/generate_sums.sh
Comment (comius) [reply proto/private/generate_sums.sh:2]: @Yannic could you add it back?

**Review by comius — CHANGES_REQUESTED** [review]

File: proto/private/dependencies.bzl
Comment (comius) [inline proto/private/dependencies.bzl:53]: This file does not exists.
Please replace with: 
https://mirror.bazel.build/github.com/protocolbuffers/protobuf/releases/download/v3.13.0/protoc-3.13.0-linux-s390x.zip
https://github.com/protocolbuffers/protobuf/releases/download/v3.13.0/protoc-3.13.0-linux-s390x.zip
and
shasum "11f95a50e193eb3d423240bc84a0a0f94b6e2043e819ba0bebcf24c7feaa3cde"

File: proto/private/dependencies.bzl
Comment (comius) [inline proto/private/dependencies.bzl:75]: Please remove, OSX 32 bit version no longer exists.

File: proto/private/generate_sums.sh
Comment (comius) [inline proto/private/generate_sums.sh:26]: Remove _64

File: proto/private/generate_sums.sh
Comment (comius) [inline proto/private/generate_sums.sh:29]: Remove.

**Review by comius — CHANGES_REQUESTED** [review]
Hey @Yannic, thanks for your effort and sorry for waiting for a review. 

I checked all files and checksums, couple of fixes are needed. I already uploaded protoc-3.13.0-linux-s390x.zip to the bazel mirror.

File: proto/private/generate_sums.sh
Comment (Yannic) [reply proto/private/generate_sums.sh]: `/usr/bin/bash` definitely doesn't exist on macOS (10.14 & 10.15), so leaving it as it is makes the script unusable for me. Replaced it with `#!/usr/bin/env bash`.

**Review by Yannic — COMMENTED** [review]
Thanks! I fixed the script to fail if urls don't exist instead of silently accepting it. PTAL.

**Review by comius — APPROVED** [review]

**Comment by Yannic** [comment]
@comius Thanks for the review! Can you also merge (I don't have write access to this repo)

**Pull request merged**
